### DISCUSSION TITLE
Get first item of `query`

### DISCUSSION THREAD

**Discussion by lettuceDestroyer** [typst/typst#3976] [Q&A] [answered]
I'm trying to set up a template which requires me to make a page break before the first heading.

What I was hoping to achieve was something like this:

```typst
  let headings = context query(heading.where(level: 1))
  show headings.at(0): it => pagebreak(weak: true) + it
```

but I always get this error message `expected string, found integer`

**Comment by laurmaedje** [comment]
You're getting the error because the output of the `context` block is opaque content and not the array. The array is only available within the context block.

But even if you put the whole show rule into the context, it's simply not possible currently to write a show rule based on a query result like this. You'd get an error like "this selector cannot be used with show".

Maybe you can write the template in a simpler way and just include a pagebreak somewhere at a known position?

**Comment by laurmaedje** [comment, marked as answer]
Here's an alternative solution I just thought of:
```typ
#show heading: it => {
  if query(heading).first().location() == it.location() {
    pagebreak()
  }
  it
}

= Introduction
= World
```

**Reply by lettuceDestroyer** [reply]
That works, thank you 😸
